tests/models/test_openai.py

```diff
@@ -482,11 +483,40 @@ async def test_no_delta(allow_model_requests: None):
         text_chunk('world'),
     )
     mock_client = MockOpenAI.create_mock_stream(stream)
-    m = OpenAIModel('gpt-4', openai_client=mock_client)
+    m = OpenAIModel('gpt-4o', openai_client=mock_client)
     agent = Agent(m)
 
     async with agent.run_stream('') as result:
         assert not result.is_complete
         assert [c async for c in result.stream_text(debounce_by=None)] == snapshot(['hello ', 'hello world'])
         assert result.is_complete
         assert result.usage() == snapshot(Usage(requests=1, request_tokens=6, response_tokens=3, total_tokens=9))
+
+
+@pytest.mark.parametrize('system_prompt_role', ['system', 'developer', None])
+async def test_system_prompt_role(
+    allow_model_requests: None, system_prompt_role: OpenAISystemPromptRole | None
+) -> None:
+    """Testing the system prompt role for OpenAI models is properly set / inferred.
+
+    It's not ideal that we test a private method here, this test is bound to be inherently fragile.
+    That beging said, inference / configurabilty here is likely to change, so we just want to test
+    the current desired behavior for now."""
+    m = OpenAIModel('gpt-4o', system_prompt_role=system_prompt_role)
+    assert m.system_prompt_role == system_prompt_role
+
+    agent_model: OpenAIAgentModel = cast(
+        OpenAIAgentModel, await m.agent_model(function_tools=[], allow_text_result=True, result_tools=[])
+    )
+    openai_messages = list(
+        chain(
+            agent_model._map_message(  # pyright: ignore[reportPrivateUsage]
+                ModelRequest(kind='request', parts=[SystemPromptPart(content='system directions')])
+            )
+        )
+    )
+
+    if system_prompt_role == 'developer':
+        assert openai_messages[0]['role'] == 'developer'
+    else:
+        assert openai_messages[0]['role'] == 'system'
```

Comment:
We should change the test to actually check all `openai_messages`, the fact that you are only asserting on the first one is why this test didn't catch the issue on my suggestion above